db/sqldb/initiative_invitation.go

```diff
@@ -0,0 +1,153 @@
+package sqldb
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/RMI/pacta/db"
+	"github.com/RMI/pacta/pacta"
+	"github.com/jackc/pgtype"
+	"github.com/jackc/pgx/v4"
+)
+
+const initiativeInvitiationIDNamespace = "iivte"
+const initiativeIntivtationSelectColumns = `id, created_at, used_at, initiative_id, used_by_user_id`
+
+func (d *DB) InitiativeInvitation(tx db.Tx, id pacta.InitiativeInvitationID) (*pacta.InitiativeInvitation, error) {
+	rows, err := d.query(tx, `
+		SELECT `+initiativeIntivtationSelectColumns+`
+		FROM initiative_invitation 
+		WHERE id = $1;`, id)
+	if err != nil {
+		return nil, fmt.Errorf("querying initiative_invitation: %w", err)
+	}
+	iis, err := rowsToInitiativeInvitations(rows)
+	if err != nil {
+		return nil, fmt.Errorf("translating rows to initiative_invitations: %w", err)
+	}
+	return exactlyOne("initiative_invitation", id, iis)
+}
+
+func (d *DB) InitiativeInvitationsByInitiative(tx db.Tx, iid pacta.InitiativeID) ([]*pacta.InitiativeInvitation, error) {
+	rows, err := d.query(tx, `
+		SELECT `+initiativeIntivtationSelectColumns+`
+		FROM initiative_invitation 
+		WHERE initiative_id = $1;`, iid)
+	if err != nil {
+		return nil, fmt.Errorf("querying initiative_invitation: %w", err)
+	}
+	iis, err := rowsToInitiativeInvitations(rows)
+	if err != nil {
+		return nil, fmt.Errorf("translating rows to initiative_invitations: %w", err)
+	}
+	return iis, nil
+}
+
+func (d *DB) CreateInitiativeInvitation(tx db.Tx, ii *pacta.InitiativeInvitation) (pacta.InitiativeInvitationID, error) {
+	if err := validateInitiativeInvitationForCreation(ii); err != nil {
+		return "", fmt.Errorf("validating initiative_invitation for creation: %w", err)
+	}
+	// Unlike other entities, II's can have user-set IDs so as to appear semantically meaningful.
+	if ii.ID == "" {
+		ii.ID = pacta.InitiativeInvitationID(d.randomID(initiativeInvitiationIDNamespace))
+	}
+	createdAt := time.Now()
```

Comment:
This can't be mocked out for tests, I'd either:

1) Set `created_at DEFAULT CURRENT_TIME` in the schema (my recommendation)
2) just use `CURRENT_TIMESTAMP` or `NOW` directly in the SQL here
3) have a `d.now()` func on the DB that can be mocked out, which we've done in other projects.

Reply:
Oh sounds good to me!! I thought you didn't like `DEFAULT X` on creation!?

Reply:
I can't think of any reason I wouldn't like it, it seems pretty footgun-free to me on a correctly configured database